### PR TITLE
Sanitize http method to uppercase

### PR DESCRIPTION
**Pull request by glefloch**
This pull request aims to close issue #387. As this is part of version 0.10.0, should it be merge in master or in another branch? 

Signed-off-by: glefloch <glfloch@gmail.com>

### CODE REVIEW

**Comment by glefloch** [comment]
Yes for sure, no problem

**Comment by GheRivero** [comment]
LGTM. small nit

**Comment by beorn7** [comment]
I changed the base branch and will merge now. Thanks.

File: prometheus/promhttp/instrument_server.go [hunk 1]
```diff
@@ -322,23 +322,23 @@ func computeApproximateRequestSize(r *http.Request) int {
...
+		return "CONNECT"
 	case "OPTIONS", "options":
-		return "options"
+		return "OPTIONS"
 	case "NOTIFY", "notify":
-		return "notify"
+		return "NOTIFY"
 	default:
-		return strings.ToLower(m)
+		return strings.ToUpper(m)
```
Comment (GheRivero) [inline prometheus/promhttp/instrument_server.go]: The default rule will work for any case. No need to duplicate code. 
Or made the default case to error: non permitted HTTP method.

**Review by GheRivero — COMMENTED** [review]
.

File: prometheus/promhttp/instrument_server.go (hunk 1 above)
Comment (beorn7) [reply prometheus/promhttp/instrument_server.go]: > The default rule will work for any case. No need to duplicate code.

I don't understand. What code is duplicated here?

> Or made the default case to error: non permitted HTTP method.

The goal here is not to validate. If people really want to, they could come up with fantasy HTTP methods (or there might even be more added officially). The goal here is to make sure that the method is always upper case.

Comment (GheRivero) [reply prometheus/promhttp/instrument_server.go]: If you remove all the cases, except the default one, the output will be the same. Or even more simple:
```
func sanitizeMethod(m string) string {
    return strings.ToUpper(m)
}
```

Comment (GheRivero) [reply prometheus/promhttp/instrument_server.go]: Anyway, it's just a cosmetic suggestion for clarity, so just ignore it if it's not worth the time.

**Pull request merged**